### DISCUSSION TITLE
Financial Data TTM?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [ranaroussi/yfinance#1596] [General]
Hello - is there any way to get TTM-data from the Financials informations?

I tried the actual version with this code

```
import yfinance as yf
ticker = "AAPL"
dataYF = yf.Ticker(ticker)
print(dataYF.financials)
```

But with that i only get the informations starting from SEP back but not the TTM results:

```
                                                        2022-09-30      2021-09-30      2020-09-30      2019-09-30
Tax Effect Of Unusual Items                                    0.0             0.0             0.0             0.0
Tax Rate For Calcs                                        0.162045        0.133023        0.144282        0.159438
Normalized EBITDA                                   130541000000.0  120233000000.0   77344000000.0   76477000000.0
Net Income From Continuing Operation Net Minori...   99803000000.0   94680000000.0   57411000000.0   55256000000.0
Reconciled Depreciation                              11104000000.0   11284000000.0   11056000000.0   12547000000.0
Reconciled Cost Of Revenue                          223546000000.0  212981000000.0  169559000000.0  161782000000.0
```

**Comment by ValueRaider** [comment]
https://github.com/ranaroussi/yfinance/issues/1345#issuecomment-1407435411

**Comment by init-js** [comment]
Here's one version of it. I needed the TTM for a custom purpose.

https://github.com/ranaroussi/yfinance/pull/1643

I was expecting the TTM on the website to be the sum of the last 4 quarterly reports, but it's not quite right. I suspect that there's some accounting special sauce added to the calculation, or some linear interpolation action going on at finance yahoo com.

(i recognize that summing the numbers from the last 4 quarterly reports makes sense for some keys which are additive (revenues), but not for others, (like ratios), of course)

**Comment by shuaiwang88** [comment]
is it possible to get more than 4 quarters of data?
